python/lsst/ts/MTAlignment/alignmentCSC.py

```diff
@@ -0,0 +1,198 @@
+    ):
+        super().__init__(
+            "MTAlignment",
+            index=0,
+            config_schema=CONFIG_SCHEMA,
+            config_dir=config_dir,
+            initial_state=initial_state,
+            simulation_mode=simulation_mode,
+        )
+        self.model = None
+        self.max_iters = 3
+
+        # temporary variables for position; these will eventually be supplied
+        # by the TMA and camera rotator CSCs.
+        self.elevation = 90
+        self.azimuth = 0
+        self.camrot = 0
+
+    async def handle_summary_state(self):
+        if self.disabled_or_enabled:
+            if self.model is None:
+                self.model = AlignmentModel(self.config.t2sa_ip, self.config.t2sa_port)
+        else:
+            self.model = None
+
+    async def configure(self, config):
+        self.config = config
+
+    @staticmethod
+    async def get_config_pkg(self):
+        return "ts_config_mttcs"
+
+    async def do_measureTarget(self):
+        """Measure and return coordinates of a target. Options are
+        M1M3, M2, CAM, and DOME"""
+        pass
+
+    async def do_align(self):
+        """Perform correction loop"""
+        pass
+
+    async def do_healthCheck(self):
+        """run healthcheck script"""
+        pass
+
+    async def do_laserPower(self):
+        """put the laser in sleep state"""
+        pass
+
+    async def do_powerOff(self):
+        """full power off of tracker and interface"""
+        pass
+
+    async def do_measurePoint(self):
+        """measure and return coords of a specific point"""
+        pass
+
+    async def do_pointDelta(self):
+        """return a vector between two points"""
+        pass
+
+    async def do_setReferenceGroup(self):
+        """Nominal point group to locate tracker station to and provide data
+        relative to"""
+        pass
+
+    async def do_setWorkingFrame(self):
+        """attempt to set the passed string as the SpatialAnalyzer working
+        frame"""
+        pass
+
+    async def do_halt(self):
+        """halts any executing measurement plan and returns to ready state"""
+        pass
+
+    async def do_loadSATemplateFile(self):
+        """SA Template file path and name. This is in the filesystem on the
+        T2SA host."""
+        pass
+
+    async def do_measureDrift(self):
+        """measure tracker drift"""
+        pass
+
+    async def do_resetT2SA(self):
+        """reboots t2sa and SA"""
+        pass
+
+    async def do_newStation(self):
+        """create new tracker station"""
+        pass
+
+    async def do_saveJobfile(self):
+        """save job file"""
+        pass
+
+    async def correction_loop(self):
+        await self.model.set_telescope_position(
+            self.elevation, self.azimuth, self.camrot
+        )
+
+        await self.model.measure_m1m3()
+        aligned = False
+        loopcount = 1
+        while not aligned:
```

Comment:
Out of curiosity, how does `aligned` get updated? It looks like the loop will always take three iterations and then exists. Will this be implemented in a future code push?

Reply:
It doesn't, really. I originally wrote this class  last march, when we had the laser tracker set up in the basement and we wanted to test the API between the vendor software and my model.  I just wanted to drive the hardware through an approximation of what an actual alignment-correction loop would look like. I only left this code in so that the notebooks will still work, because we're planning to do some more testing next month.